### DISCUSSION TITLE
v1.0.0

### DISCUSSION THREAD

**Discussion by Melvin-Abraham** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#545] [General]
# v1.0.0 Changelog

- Bumped electron version to 10.3.2 for Windows and MacOS
- Debian and RedHat packages are now available for Linux platform
- Prevent microphone from being active when not in use
- Fixed assistant error on first launch which occurred due to window positioning and desktop resolution
- Added support for _"Hey Google / Ok Google"_ wake word detection **(experimental)**
- Added proper **auto-update** functionality for certain package/app formats over multiple platforms
- Added option to configure custom assistant hotkey (keyboard shortcut)
- Added **"Troubleshoot"** submenu in the tray context menu
- Added ability to set display monitor preference for multi-monitor setup
- Added ability to select custom microphone and speaker source
- Launching the assistant from launcher now focuses the running instance automatically instead of showing an error
- Now the user will be shown token expiry error instead of the generic *"invalid_grant"* error with an option to quickly reset the access tokens
- Prevent shutdown interruption on MacOS
- Fixed most of the issues relating to **"Close on Blur"** window behavior
- Added multiple window-border options to create separation between the app and background
- Fixed issue where clicking the suggestion chip did not respond
- Assistant can now infer _supported_ assistant language based on system preferences
- Improvements to assistant display languages _(special thanks to the community)_
- Use up/down arrow keys on **"Ask me anything..."** box to flip through history
- Added option to use the <kbd>Esc</kbd> key to close or minimize the assistant window
- Now the typed queries by default do not reply the user back via voice (behavior can be customized)
- Added "Feedback & Links" section in settings for helpful links
- Consistent notification display on all platforms during initial app launch
- Warn user when "savedTokensPath" is not writable (probably due to permission)

Other minor changes
-------------------

- Added support for copying authentication link and retrying in the **"Get Token"** screen
- Prevent token saving when initially starting app as hidden
- Added command line verbose logging for debugging purposes
- Prevent triggering microphone when assistant is not initialized
- Added option to toggle sending notification on startup
- Added option to toggle hiding the assistant on first launch
- Minor consistency in display language
- Other minor enhancements and bug fixes

<hr /><em>This discussion was created from the release <a href='https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/releases/tag/'>v1.0.0</a>.</em>

**Comment by sumeshir26** [comment]
Will there be more frequent releases from now? I think 1year is a bit too much

**Reply by Melvin-Abraham** [reply]
Probably yes since the stable version has been released and any upcoming updates would be minor but as far as code rewriting is concerned, it might take longer.

**Reply by im-coder-lg** [reply]
Will there be release candidates to 2.0.0? Or is it 1.0.1 and so on?

**Reply by Melvin-Abraham** [reply]
There will be minor releases like `1.x.y` while continuing development on the next major version, `2.0.0` . So, there could be release candidates for the same later on...

**Reply by Arplane-SOCITY** [reply]
# Google Assistant [Unofficial] Is Have Thai/ไทย?
# Example:
- Say "Ok Google, อรุณสวัสด์"

**Reply by iamneo99** [reply]
Hello all, can we have build for raspberry pi or the arm64 version please?

**Comment by kreislerramirezmt** [comment]
Nice <3

**Comment by smoliicek** [comment]
What font did it use?

**Reply by im-coder-lg** [reply]
I think Google Sans and etc... Gotta look on that later.

**Reply by sumeshir26** [reply]
@Smajlll @im-coder-lg Product Sans

**Comment by Nicus009** [comment]
Please add russian language

**Comment by addy118** [comment]
which path we have to put in key path

**Reply by im-coder-lg** [reply]
From the Google Cloud Platform, you would download a JSON file, right? Save it to an undisclosed location in your computer that you only know and enter that location's path in the Key Path.

**Comment by yunidotcom** [comment]
Please add Indonesian language

**Reply by im-coder-lg** [reply]
If the Google Assistant SDK supports it, we'll definetely add it!

**Comment by SamipyaMainali** [comment]
Is there any way I can control Spotify and Netflix and/or even YouTube with it.

**Reply by sumeshir26** [reply]
Not yet

**Comment by achalmed** [comment]
genial!!!

**Reply by im-coder-lg** [reply]
What do you mean?

**Reply by EkCaAv** [reply]
> What do you mean?

Very Good!

**Comment by freefloater** [comment]
I hope to have Chinese mode, thank you.
希望可以有中文模式謝謝。

**Reply by Art657** [reply]
中文！

**Reply by AiixiBear** [reply]
me too 我也要中文謝謝

**Comment by blackstar123** [comment]
why not finish an youtube application for windows lattop(win10)  , not just visit Youtube from web from browser(I hate this kind),please

**Reply by im-coder-lg** [reply]
Anyway, you would have the YouTube Web experience, if we use Electron. A client for PCs would be a bit overloaded, right?

**Comment by EsaulTellez** [comment]
I have this problem
make sure the file is of the form "client_secret_<your_id>.apps.googleusercontent.com.json
why? help me plsssss

**Reply by EkCaAv** [reply]
This happens to me too! :(

**Reply by Tbnrfps42** [reply]
use this link
https://www.youtube.com/watch?v=fZ3EKCxTgi4

**Comment by sumeshir26** [comment]
@EsaulTellez #671

**Comment by EkCaAv** [comment]
Hello! There was an error choosing the JSON file and whenever I try to open the file, and restart, I get this message: 
![image](https://user-images.githubusercontent.com/80602430/181116136-fdca67b6-13b1-4178-87cc-fb626e35bf5e.png)
I don't know what I should do, I already reinstalled and did all the steps again and nothing works for me :/.

**Reply by im-coder-lg** [reply]
#671, you checked it, right?

**Comment by Mihlayonke** [comment]
Hi,

I did all the steps on consoles but now I can`t download the Google Assistant Setup, the browser error says " Couldn`t downloaded-need authorization"

![image](https://user-images.githubusercontent.com/65561989/181876789-898cd6b3-ab0a-4420-8244-07e87f8ad518.png)

**Reply by ghost** [reply]
I'm  not sure what's going on, but try downloading the source code perhaps.

**Reply by im-coder-lg** [reply]
Do you have the permission to download in that folder? Perhaps, you could try building an installer from the source code, it adds a bit more fixes.

Firstly, try switching browsers, if you don't want to build from source code.
